src/test/java/org/jenkinsci/plugins/pipeline/utility/steps/conf/WriteYamlStepTest.java

```diff
@@ -0,0 +1,129 @@
+package org.jenkinsci.plugins.pipeline.utility.steps.conf;
+
+import hudson.model.Label;
+import hudson.model.Result;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.jvnet.hudson.test.JenkinsRule;
+
+public class WriteYamlStepTest {
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+    @Rule
+    public TemporaryFolder temp = new TemporaryFolder();
+
+    private final String list =  "[['ELASTICSEARCH':["
+            + "'image': 'elasticsearch:5.2.0',"
+            + "'sleep': 30,"
+            + "'healthcheck': 9300,"
+            + "'cmd': 'elasticsearch -Ecluster.name=%%JUID -Enetwork.host=0.0.0.0 -Ediscovery.zen.minimum_master_nodes=1',"
+            + "'env': ['ES_JAVA_OPTS=\"-Des.cluster.name=%%JUID -Des.network.host=%%OWNHOSTNAME\"']]]]";
+
+    @Before
+    public void setup() throws Exception {
+        j.createOnlineSlave(Label.get("slaves"));
+    }
+
+    @Test
+    public void writeInvalidMap() throws Exception {
+        WorkflowJob p = j.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition(
+                "def l = [] \n" +
+                        "node('slaves') {\n" +
+                        "  writeYaml file: 'test', data: /['a': ]/ \n" +
+                        " sh 'cat test' \n" +
```

Comment:
There is a feature coming soon to pipeline which stores the parameters of steps, that might risk printing the step to the log, which means your assert might not assert on the thing it think it does.

It would be more precise if you archived the file and read it back in in the test to verify the contents. But it's just a _nit_.